integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/rule/RowCountRuleImpl.java

```diff
@@ -0,0 +1,119 @@
+package com.thinkbiganalytics.spark.dataquality.rule;
+
+/*-
+ * #%L
+ * kylo-spark-job-dataquality-app
+ * %%
+ * Copyright (C) 2017 ThinkBig Analytics
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.thinkbiganalytics.spark.dataquality.util.DataQualityConstants;
+import com.thinkbiganalytics.spark.dataquality.util.FlowAttributes;
+import com.thinkbiganalytics.spark.dataquality.util.MissingAttributeException;
+
+/**
+ * Data Quality Rule to ensure that: <br>
+ * Number of Source rows = Valid row count + Invalid row count
+ */
+public class RowCountRuleImpl implements DataQualityRule {
+    private static final Logger log = LoggerFactory.getLogger(RowCountRuleImpl.class);
+
+    private String name;
+    private String description;
+    private boolean status;
+
+    long sourceRowCount;
+    long invalidRowCount;
+    long validRowCount;
+
+    public RowCountRuleImpl() {
+        this.name = "ROW_COUNT_TOTAL_RULE";
+        this.description = "Source Row Count == (Valid and Invalid Row Counts)";
```

Comment:
Description can be made more verbose to be consistent with other rule descriptions.